mock-bank-system/src/main/java/com/centiglobe/mockbanksystem/presentation/BankController.java

```diff
@@ -0,0 +1,108 @@
+package com.centiglobe.mockbanksystem.presentation;
+
+import org.slf4j.LoggerFactory;
+
+import com.prowidesoftware.swift.model.mx.AbstractMX;
+import com.prowidesoftware.swift.model.mx.BusinessAppHdrV02;
+import com.prowidesoftware.swift.model.mx.MxPacs00800109;
+import com.prowidesoftware.swift.model.mx.dic.CreditTransferTransaction43;
+
+import org.slf4j.Logger;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * A controller for handling ISO 20022 message
+ * from the external component
+ * 
+ * @author William Stackenäs
+ */
+@RestController
+public class BankController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BankController.class);
+
+    private static double balance = 0;
+
+    private static String pacs002 =
+    "<?xml version=\"1.0\" encoding=\"UTF-8\" ?>" +
+    "<RequestPayload>" +
+    "    <h:AppHdr xmlns:h=\"urn:iso:std:iso:20022:tech:xsd:head.001.001.02\">" +
+    "        <h:Fr>" +
+    "            <h:FIId>" +
+    "                <h:FinInstnId>" +
+    "                    <h:Nm>%s</h:Nm>" +
+    "                </h:FinInstnId>" +
+    "            </h:FIId>" +

```

Comment:
We might want to generate the pacs002 messages on the fly to create a unique `pacs.002`. This important if we want to be more proper because of elements such as creation date-time and message id.

Reply:
I agree that it would be for the best, but this is a simple mock bank system meant for demo purposes that doesn't even check which currency the payment it receives is in, it simply increments its internal wallet by whatever "amount" of money it receives. 😆 In case we add functionality for validating the pacs002 response in the communication system I think it is fine as it is, what about you?

Reply:
I think it is fine as-is for now and if we have time maybe we should make it a bit more advanced. 👌